### ISSUE TITLE
pacman 6 fails to handle DB signatures if they get redirected to a different basename

### ISSUE THREAD

**Issue by lazka** [msys2/msys2-pacman#4]
pacman fails to sync DB signatures:

```
$ curl -I https://mirror.msys2.org/mingw/clang32/clang32.db.sig
HTTP/2 302
cache-control: private, no-cache
content-type: text/html; charset=utf-8
date: Wed, 16 Jun 2021 19:06:06 GMT
link: <https://repo.msys2.org/mingw/clang32/clang32.db.tar.gz.sig>; rel=duplicate; pri=1; geo=de
location: https://mirror.yandex.ru/mirrors/msys2/mingw/clang32/clang32.db.tar.gz.sig
server: Mirrorbits/v0.5.1-31-geeea0e0-dirty
```

This makes pacman write out a "clang32.db.tar.gz.sig" file and then it errors out because it can't find "clang32.db.sig"

```
debug: ucrt64.db: url is https://mirror.msys2.org/mingw/ucrt64/ucrt64.db
debug: ucrt64.db: maxsize 134217728
debug: ucrt64.db: opened tempfile for download: /var/lib/pacman/sync/ucrt64.db.part (wb)
debug: ucrt64.db: curl returned result 0 from transfer
debug: ucrt64.db: response code 200
debug: ucrt64.db.tar.gz.sig: url is https://repo.msys2.org/mingw/ucrt64/ucrt64.db.tar.gz.sig
debug: ucrt64.db.tar.gz.sig: maxsize 16384
debug: ucrt64.db.tar.gz.sig: opened tempfile for download: /var/lib/pacman/sync/ucrt64.db.tar.gz.sig.part (wb)
debug: ucrt64.db.tar.gz.sig: curl returned result 0 from transfer
debug: ucrt64.db.tar.gz.sig: response code 200
...
debug: "/var/lib/pacman/sync/ucrt64.db.sig" is not readable: No such file or directory
debug: sig path /var/lib/pacman/sync/ucrt64.db.sig could not be opened
debug: got error 43 at _alpm_gpgme_checksig (../lib/libalpm/signing.c: 593) : missing PGP signature
debug: missing required signature
error: ucrt64: missing required signature
debug: failed to validate db: ucrt64
```

worked with 5.2:

```
debug: url: https://mirror.msys2.org/mingw/clang64/clang64.db
debug: maxsize: 134217728
debug: opened tempfile for download: /var/lib/pacman/sync/clang64.db.part (wb)
downloading clang64.db...
debug: curl returned error 0 from transfer
debug: response code: 200
debug: url: https://mirror.msys2.org/mingw/clang64/clang64.db.sig
debug: maxsize: 16384
debug: opened tempfile for download: /var/lib/pacman/sync/clang64.db.sig.part (wb)
downloading clang64.db.sig...
debug: curl returned error 0 from transfer
debug: response code: 200
debug: checking signature for /var/lib/pacman/sync/clang64.db
debug: 1 signatures returned
debug: fingerprint: 5F944B027F7FE2091985AA2EFA11531AA0AA7F57
```

**Comment by lazka** [comment]
This did work with pacman 5

Two things to do:

* file a bug with pacman
* ~don't deref symlinks: https://github.com/etix/mirrorbits/blob/eeea0e026d0ca99c7009a91a54369f7c29e87049/filesystem/fs.go#L33 since the problem (pacman 6) is already out there.~ not worth it..

**Comment by lazka** [comment]
pacman bug filed here: https://bugs.archlinux.org/task/71274

**Comment by jeremyd2019** [comment]
Proposed upstream patch: https://lists.archlinux.org/pipermail/pacman-dev/2021-June/025231.html

**Comment by jeremyd2019** [comment]
Looks like this was pushed: https://gitlab.archlinux.org/pacman/pacman/-/commit/0147de169a2abd193699957d4e76aec522901fd2

**Comment by lazka** [comment]
fixed in 6.0.1
